infrastructure/network/utils.go

```diff
@@ -0,0 +1,64 @@
+package network
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/vishvananda/netlink"
+	"github.com/weaveworks/flintlock/core/models"
+)
+
+const (
+	ifaceLength       = 7
+	retryGenerate     = 5
+	randomBytesLength = 32
+	prefix            = "fl"
+	tapPrefix         = "tap"
+	// It's vtap only to save space.
+	macvtapPrefix = "vtap"
+)
+
+func generateRandomName(prefix string) (string, error) {
+	id := make([]byte, randomBytesLength)
+	if _, err := io.ReadFull(rand.Reader, id); err != nil {
+		return "", err
+	}
+
+	return prefix + hex.EncodeToString(id)[:ifaceLength], nil
+}
+
+func NewIfaceName(ifaceType models.IfaceType) (string, error) {
+	devPrefix := ""
+
+	switch ifaceType {
+	case models.IfaceTypeTap:
+		devPrefix = fmt.Sprintf("%s%s", prefix, tapPrefix)
+	case models.IfaceTypeMacvtap:
+		devPrefix = fmt.Sprintf("%s%s", prefix, macvtapPrefix)
+	case models.IfaceTypeUnsupported:
+		return "", interfaceErrorf("unsupported interface type: %s", ifaceType)
+	default:
+		return "", interfaceErrorf("unsupported interface type: %s", ifaceType)
```

Comment:
why have 2 returns for the same case here?

Reply:
The first one for explicit value, the second is a safe-guard if new type is added, catch that too.

Reply:
updated the second one now say "unknown" not "unsupported"